### PR TITLE
Initial project structure

### PR DESCRIPTION
**Pull request by BraydonKains**
* Created independent data, graphql, and auth packages
* Moved go.mod to the root of the project
* Added Makefile to build GraphQL server
* Added .github folder where CI stuff will go
* Moved server type into its own file
* Added User data tests

### CODE REVIEW

**Comment by Matt-Hurd** [comment]
LGTM on mobile, would like to get on a dev machine and try to run this before merging.

File: go-rest-example/api/router.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"github.com/gorilla/mux"
+)
+
+func LoadRouter() *mux.Router {
+	router := mux.NewRouter()
+
+	loadPlatformRoutes(router)
+	// loadGameRoutes(router)
```
Comment (zdylag) [inline go-rest-example/api/router.go]: nit: Can we remove this?

File: go-rest-example/auth/auth.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package auth
+
+// FIXME lol
```
Comment (zdylag) [inline go-rest-example/auth/auth.go]: nit for this project: Can we make comments as complete sentences? 
eg.
> FIXME: This is not very secure.

Rational:
> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared. 
https://golang.org/doc/effective_go

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.PHONY: graphql
+gql:
+	go build ./graphql/main.go
```
Comment (zdylag) [inline Makefile:3]: nit: Newline.

**Comment by BraydonKains** [comment]
When you get around to it, the interesting things to run will be:

* open in an editor with a go plugin
* `go mod install` or `go mod vendor` to install dependencies
* `make gql` to build the graphql server (provided you have `make` installed)
* `go test ./graphql` to run the tests

**Review by zdylag — COMMENTED** [review]
Overall the idea here seems okay to me. Would you mind cleaning up the commit structure though? I think the last commit explicitly lists ~4 commits of actions.

**Comment by zdylag** [comment]
@Matt-Hurd do we want to also ask this to rebase on main instead of having the merge commit?

File: go-rest-example/auth/auth.go (hunk 2 above)
Comment (BraydonKains) [reply go-rest-example/auth/auth.go]: Yes that's fine. This file doesn't exist anymore and was part of the POC I did for rest API, so it wasn't meant for merging or anything

**Comment by zdylag** [comment]
@BraydonKains we haven't really established a "style guide" for commits so I don't want to block on them. But how does the format described by https://tip.golang.org/doc/contribute#commit_messages look to you?

**Comment by BraydonKains** [comment]
> 
> 
> Overall the idea here seems okay to me. Would you mind cleaning up the commit structure though? I think the last commit explicitly lists ~4 commits of actions.

Should I fork again from main and rewrite the changes?

**Comment by BraydonKains** [comment]
I'm going to rewrite this PR off a clean fork to avoid confusion with the commit structure and adhere to the Go convention for commit messages.

**Pull request closed without merging**